src/validation.cpp

```diff
@@ -2811,7 +2812,7 @@ bool CChainState::InvalidateBlock(CValidationState& state, const CChainParams& c
         bool ret = DisconnectTip(state, chainparams, &disconnectpool);
         // DisconnectTip will add transactions to disconnectpool; try to add these
         // back to the mempool.
-        UpdateMempoolForReorg(disconnectpool, ret);
+        UpdateMempoolForReorg(disconnectpool, (++disconnected <= 10) && ret);
```

Comment:
Commit 3af503b4a86f3743c8660f15c0a62de0289b7d94

Care to elaborate this optimization here? Looks like it could be:
```cpp
bool ret = DisconnectTip(state, chainparams, ++disconnected <= 10 ? &disconnectpool : nullptr);
...
if (disconnected <= 10) UpdateMempoolForReorg(disconnectpool, ret)
```

Reply:
No, we still need to call UpdateMempoolForReorg, even if we don't want to add the disconnected transactions back to the mempool.